### DISCUSSION TITLE
ikvdb_export_stream()

### DISCUSSION THREAD

**Discussion by victorstewart** [hse-project/hse#25] [Ideas]
would you be open to a PR that instead of writing the export to a file, instead streamed it a piece at a time to a callback, so that it could be pushed out over a socket instead?

otherwise to export the kvdb, aka write it to a file on disk, your volume occupancy would have to be <50%.

also would avoid moving the data to disk, and then back, to send it over a socket (worse if you don't use `sendfile()`).

**Comment by smoyerx** [comment]
We are not yet to the point where we can start taking PRs, but we'll note this is a feature request for a future release.
